indico/modules/events/papers/operations.py

```diff
@@ -139,16 +158,17 @@ def create_paper_revision(paper, submitter, files):
                        _contribution=paper.contribution)
         pf.save(f.file)
     db.session.flush()
+    db.session.expire(revision._contribution, ['_paper_last_revision'])
```

Comment:
Is this still needed? IIRC the problem you had there was caused by the memoization of the `paper` property.